dpipe/im/grid.py

```diff
@@ -88,23 +89,44 @@ def build(self) -> np.ndarray:
 
 
 class Average(PatchCombiner):
-    def __init__(self, shape: Tuple[int, ...], dtype: np.dtype, **imops_kwargs: dict):
+    def __init__(self, shape: Tuple[int, ...], dtype: np.dtype, use_torch: bool = False, **imops_kwargs: dict):
         super().__init__(shape, dtype)
         self._result = np.zeros(shape, dtype)
-        self._counts = np.zeros(shape, int)
+        self._counts = np.zeros(shape, np.uint8 if use_torch else int)
         self._imops_kwargs = imops_kwargs
 
+        self._use_torch = use_torch
+
     def update(self, box: Box, patch: np.ndarray):
         slc = build_slices(*box)
 
         result_slc = self._result[slc]
-        pointwise_add(result_slc, patch.astype(result_slc.dtype, copy=False), result_slc, **self._imops_kwargs)
+        if self._use_torch:
+            result_slc_torch = torch.from_numpy(result_slc)
+            patch_torch = torch.from_numpy(patch.astype(result_slc.dtype, copy=False))
+            with torch.inference_mode():
+                result_slc_torch += patch_torch
```

Comment:
Let's use `+=` or `.add_` at L108 and L116. Are there any difference between them actually?

Reply:
done